charts/pulsar/templates/bookkeeper-cluster-initialize.yaml

```
@@ -16,7 +16,7 @@
 # specific language governing permissions and limitations
 # under the License.
 #
-{{- if .Values.initialize }}
+{{- if .Release.IsInstall }}
```

I'd suggest keeping backwards compatibility. 
```suggestion
{{- if or .Release.IsInstall .Values.initialize }}
```
The benefit of this is that the metadata initialization job can be created on demand when needed. I have been using this in test environments. It's possible to clear the persistent state of the cluster by scaling stateful sets to 0, deleting PVCs and doing a deployment with `--set initialize=true`.